src/software/ai/passing/cost_function.cpp

```diff
@@ -159,68 +208,56 @@ double calculateInterceptRisk(const Team& enemy_team, const Pass& pass,
-
-    // If the enemy cannot intercept the pass at BOTH the closest point on the pass and
-    // the receiver point for the pass, then it is guaranteed that it will not be
-    // able to intercept the pass anywhere.
+    // We estimate the intercept by the risk that the enemy robot will get to the closest
+    // point on the pass before the ball
 
-    // Figure out how long the enemy robot and ball will take to reach the closest
-    // point on the pass to the enemy's current position
-    Point closest_point_on_pass_to_robot = closestPoint(
+    Point closest_interception_point = closestPoint(
         enemy_robot.position(), Segment(pass.passerPoint(), pass.receiverPoint()));
-    // Subtracting by the max robot radius as a threshold of the enemy robot blocking the
-    // pass
-    double distance = (closest_point_on_pass_to_robot - enemy_robot.position()).length() -
-                      ROBOT_MAX_RADIUS_METERS;
-    Duration enemy_robot_time_to_closest_pass_point =
-        getTimeToTravelDistance(distance, ENEMY_ROBOT_MAX_SPEED_METERS_PER_SECOND,
-                                ENEMY_ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED);
-    Duration ball_time_to_closest_pass_point = Duration::fromSeconds(
-        (closest_point_on_pass_to_robot - pass.passerPoint()).length() / pass.speed());
-
-    // Check for division by 0
-    if (pass.speed() == 0)
-    {
-        ball_time_to_closest_pass_point =
-            Duration::fromSeconds(std::numeric_limits<int>::max());
-    }
-
-    // Figure out how long the enemy robot and ball will take to reach the receive point
-    // for the pass.
-    Duration enemy_robot_time_to_pass_receive_position =
-        enemy_robot.getTimeToPosition(pass.receiverPoint());
-    Duration ball_time_to_pass_receive_position = pass.estimatePassDuration();
-
-    double robot_ball_time_diff_at_closest_pass_point =
-        ((enemy_robot_time_to_closest_pass_point + enemy_reaction_time) -
-         (ball_time_to_closest_pass_point))
-            .toSeconds();
-    double robot_ball_time_diff_at_pass_receive_point =
-        ((enemy_robot_time_to_pass_receive_position + enemy_reaction_time) -
-         (ball_time_to_pass_receive_position))
-            .toSeconds();
-
-    double min_time_diff = std::min(robot_ball_time_diff_at_closest_pass_point,
-                                    robot_ball_time_diff_at_pass_receive_point);
+    Vector enemy_interception_vector =
+        closest_interception_point - enemy_robot.position();
+    // Take into account the enemy robot's radius for minimum min_interception_distance
+    // required to travel to intercept the pass.
+    double min_interception_distance = std::max(
+        0.0, enemy_interception_vector.length() -
+                 ROBOT_MAX_RADIUS_METERS);  // TODO (NIMA): It is potentially faster to
+                                            // travel to +radius than -radius
+
+    const double ENEMY_ROBOT_INTERCEPTION_SPEED_METERS_PER_SECOND = 0.5;
```

Comment:
why'd you add this constant here

Reply:
I just moved this code. It was there from before. At some point we could try improving keep away.